libs/SmartSync/src/com/salesforce/androidsdk/smartsync/util/SyncUpTarget.java

```diff
@@ -0,0 +1,200 @@
+        JSONObject target = new JSONObject();
+        target.put(ANDROID_IMPL, getClass().getName());
+        return target;
+    }
+
+    /**
+     * Save locally created record back to server
+     * @param syncManager
+     * @param objectType
+     * @param fields
+     * @return server record id or null if creation failed
+     * @throws JSONException
+     * @throws IOException
+     */
+    public String createOnServer(SyncManager syncManager, String objectType, Map<String, Object> fields) throws JSONException, IOException {
+        RestRequest request = RestRequest.getRequestForCreate(syncManager.apiVersion, objectType, fields);
+        RestResponse response = syncManager.sendSyncWithSmartSyncUserAgent(request);
+
+        return response.isSuccess()
+                ? response.asJSONObject().getString(Constants.LID)
+                : null;
+    }
+
+    /**
+     * Delete locally deleted record from server
+     * @param syncManager
+     * @param objectType
+     * @param objectId
+     * @return true if successful
+     * @throws JSONException
+     * @throws IOException
+     */
+    public boolean deleteOnServer(SyncManager syncManager, String objectType, String objectId) throws JSONException, IOException {
+        RestRequest request = RestRequest.getRequestForDelete(syncManager.apiVersion, objectType, objectId);
+        RestResponse response = syncManager.sendSyncWithSmartSyncUserAgent(request);
+
+        return response.isSuccess();
+    }
+
+    /**
+     * Save locally updated record back to server
+     * @param syncManager
+     * @param objectType
+     * @param objectId
+     * @param fields
+     * @return true if successful
+     * @throws JSONException
+     * @throws IOException
+     */
+    public boolean updateOnServer(SyncManager syncManager, String objectType, String objectId, Map<String, Object> fields) throws JSONException, IOException {
+        RestRequest request = RestRequest.getRequestForUpdate(syncManager.apiVersion, objectType, objectId, fields);
+        RestResponse response = syncManager.sendSyncWithSmartSyncUserAgent(request);
+
+        return response.isSuccess();
+    }
+
+    /**
+     * Fetch last modified date for a given record
+     * @param syncManager
+     * @param objectType
+     * @param objectId
+     * @return
+     * @throws JSONException
+     * @throws IOException
+     */
+    public long fetchLastModifiedDate(SyncManager syncManager, String objectType, String objectId) throws JSONException, IOException {
+        boolean isNewer = false;
+        long serverLastModified = -1L;
+        final String query = SOQLBuilder.getInstanceWithFields(Constants.LAST_MODIFIED_DATE)
+                .from(objectType)
+                .where(Constants.ID + " = '" + objectId + "'")
+                .build();
+
+        RestResponse lastModResponse = syncManager.sendSyncWithSmartSyncUserAgent(RestRequest.getRequestForQuery(syncManager.apiVersion, query));
```

Comment:
During sync, we use synchronous network calls - but the sync itself happens on a separate thread